R/package-auth.R

```diff
@@ -88,7 +87,14 @@ db_token <- function(profile = getOption("db_profile")) {
     return(token)
   }
 
-  read_env_var(key = "token", profile = profile, error = FALSE)
+  token <- read_env_var(key = "token", profile = profile, error = FALSE)
+
+  # Checks for OAuth Databricks token inside RStudio on Posit Workbench
+  if (is.null(token) && exists(".rs.api.getDatabricksToken")) {
+    getDatabricksToken <- get(".rs.api.getDatabricksToken")
+    token <- getDatabricksToken(db_host())
+  }
+  return(token)
```

Comment:
I don't think it's a great idea to use `.rs.api.getDatabricksToken()` here. This is what we did originally in `odbc`, but it had a bunch of weird side effects like code not working in jobs or when rendering RMarkdown documents. We moved to [parsing the `DATABRICKS_CONFIG_FILE` file that Workbench generates instead](https://github.com/r-dbi/odbc/blob/ab9421fa7868a383696b35515127d2a0b87d5d42/R/driver-databricks.R#L324-L345). Since this PR already seems to support that, maybe it already works without this code?

Reply:
(Since `odbc` doesn't support parsing all `.databrickscfg` files, [we also check](https://github.com/r-dbi/odbc/blob/ab9421fa7868a383696b35515127d2a0b87d5d42/R/driver-databricks.R#L254-L258) if the path contains `posit-workbench` before we try parsing it, but you probably don't have that issue here.)

Reply:
Good call, I will remove it and focus on trigger the config file flow automatically instead which I added support in this PR